pylint/checkers/classes/class_checker.py

```diff
@@ -1359,6 +1365,38 @@ def _check_slots(self, node):
                     self._check_slots_elt(elt, node)
                 except astroid.InferenceError:
                     continue
+            self._check_redefined_slots(node, slots, values)
+
+    def _check_redefined_slots(
+        self,
+        node: nodes.ClassDef,
+        slots_node: nodes.NodeNG,
+        slots_list: List[nodes.NodeNG],
+    ) -> None:
+        """Check if `node` redefines a slot which is defined in an ancestor class"""
+        slots_names = [
+            slot.value
+            if isinstance(slot, nodes.Const)
+            else getattr(safe_infer(slot), "value", "")
+            for slot in slots_list
+        ]
```

Comment:
```suggestion
        slots_names: List[str] = []
        for slot in slots_list:
            if isinstance(slot, nodes.Const):
                slots_names.append(slot.value)
            else:
                inferred_slot = safe_infer(slot)
                if inferred_slot:
                    slots_names.append(inferred_slot.value)
```

It's a bit longer, but I prefer how explicit and readable this is. Let me know what you think!

Reply:
I'm also fine with that. Thanks @DanielNoord